addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -55,6 +55,12 @@ const clearServiceCache = () => {
         rpc: {},
     };
 };
+
+const CSS_ANIMATION_RULE_REGEX =
+    /(?<declaration>animation(?:-duration)?: .*?)(?<value>(?:\d+(?:\.\d+)?)|(?:\.\d+))(?<unit>ms|s)(?<separator>\s|;|"|$)/gm;
+const SVG_DUR_TIMECOUNT_VAL_REGEX =
+    /(?<attribute_name>\sdur="\s*)(?<value>(?:\d+(?:\.\d+)?)|(?:\.\d+))(?<unit>h|min|ms|s)?\s*"/gm;
+const CSS_ANIMATION_RATIO_REGEX = /(--animation_ratio: (?<ratio>\d*(\.\d+)?));/m;
```

Comment:
Mention in a comment that those regex are duplicated server-side. Should we have to fix them, it's better to tie everything together.

If you can also document why we need to duplicate most of the logic at the right places... :pray: